vapid/soa.h

```diff
@@ -196,21 +194,32 @@ namespace vapid {
             ((sort_col_by_reference(std::integral_constant<size_t, I>{})), ...);
         }
 
+        template <class C>
+        void reorder(C &c, std::vector<size_t> &o)
+        {
+            using std::swap;
+            size_t num_elems = o.size();
+            for (size_t i=0; i<num_elems; ++i)
+            {
```

Comment:
I think I see how this works - is this right?
```
// Loop invariant:
// 1. o is a permutation that would unscramble the vector c
// 2. o[i'] = i for all i' < i
```

Reply:
No problem, that was fun! The vector o provides the new position for each index. For example if o[0] = 3, it means that the values at index 0 should be moved to index 3 to be sorted.

So basically we first swap c[0] and c[3] (same for o). Now c[3] is at the sorted position, but c[0] has a new value which may have to go somewhere else, so we continue until o[i] == i.

I'm somewhat surprised it is slower. Maybe because writing the vector in sequence is more efficient than rearranging it with swaps. The drawback is that you need two vector,SensorData> which could be an issue if these become very large.